### PR TITLE
vimPlugins.nvim-treesitter-sexp: init at 2024-06-07

### PR DESCRIPTION
**Pull request by nwjsmith**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348811`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
LGTM, thanks

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -1669,6 +1669,18 @@ in
 
   nvim-treesitter-parsers = lib.recurseIntoAttrs self.nvim-treesitter.grammarPlugins;
 
+  nvim-treesitter-sexp = super.nvim-treesitter-sexp.overrideAttrs {
+    dependencies = with self; [
+      (nvim-treesitter.withPlugins (p: [
```
Comment (khaneliman) [inline pkgs/applications/editors/vim/plugins/overrides.nix:1674]: @GaetanLepage This actually got me thinking about our discussion in matrix.. Will we run into issues with plugins that declare their own treesitter that might conflict with other treesitters in rtp?

Comment (abhisheksingh0x558) [reply pkgs/applications/editors/vim/plugins/overrides.nix:1674]: Isn't it better to allow the user to install the grammar instead of declaring them here?

Comment (khaneliman) [reply pkgs/applications/editors/vim/plugins/overrides.nix:1674]: I guess I'd be curious if the nvimRequireCheck passes without declaring the treesitter dependency.

Comment (nwjsmith) [reply pkgs/applications/editors/vim/plugins/overrides.nix:1674]: @khaneliman it works a-okay. @abhisheksingh0x558 I'll get rid of the nvim-treesitter dep

**Review by abhisheksingh0x558 — APPROVED** [review]
LGTM

**Review by khaneliman — APPROVED** [review]
Alright, LGTM then if treesitter isn't a hard requirement on require so that we dont cause conflicts in rtp. We probably need to address a few other plugins for this.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348811`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.nvim-treesitter-sexp</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Pull request merged**